unblob/handlers/archive/rar.py

```diff
@@ -1,175 +1,46 @@
 
-        extra_data = 0
-        # LHD_LONG_BLOCK flag indicates that there's another 32-bit field after the normal
-        # base header, which tells you the size of the extra data. So in this case, we would
-        # want to skip past that extra data as well.
-        # https://gitlab.gnome.org/GNOME/evince/-/blob/d69158ecf0e2a2f1562b06c265fc86f87fe7dd6f/cut-n-paste/unarr/rar/parse-rar.c#L29
-        # This also suggests that this should be true without this flag, if the type is 0x74,
-        # i.e. a FILE_HEAD section.
-        if (
-            header.flags
-            & [x[0] for x in FILE_HEADER_FLAGS if x[1] == "LHD_LONG_BLOCK"][0]
-            or header.type == 0x74
-        ):
-            header = cparser.base_vol_header_data(file)
-            file.seek(offset)
-            extra_data = header.datasize
+def calculate_rar_size(file: io.BufferedIOBase, start_offset: int):
+    rar_limited = LimitedStartReader(file, start_offset)
+    rar_file = rarfile.RarFile(rar_limited)
+    rar_tell = rar_file._rarfile.tell()
+    # The tell points after the final byte, so the size is tell() - 1
+    return rar_tell - 1
 
-        if header.type == 0x72:  # "MARK_HEAD"
-            # This is the RAR signature (0x72 indicates MARK_HEAD)
-            # It looks like 52 61 72 21 1A 07 00, and the 0x0007 is
-            # the size field, and how long it is. Clever!
-            pass
-        elif header.type == 0x73:  # "MAIN_HEAD"
-            pass
-        elif header.type == 0x74:  # "FILE_HEAD"
-            pass
-        elif header.type == 0x75:  # "COMM_HEAD"
-            pass
-        elif header.type == 0x76:  # "AV_HEAD"
-            pass
-        elif header.type == 0x77:  # "SUB_HEAD"
-            pass
-        elif header.type == 0x78:  # "PROTECT_HEAD"
-            pass
-        elif header.type == 0x79:  # "SIGN_HEAD"
-            pass
-        elif header.type == 0x7A:  # "NEWSUB_HEAD"
-            pass
-        elif header.type == 0x7B:  # "ENDARC_HEAD"
-            # End of archive here
-            return offset + header.size
-        else:
-            # This isn't an expected RAR internal header, so we
-            # assume we've hit the end of something-like-a-RAR.
-            # It's not ending with the ENDARC_HEAD, so it may be
-            # corrupted?
-            return offset
 
-        offset = offset + header.size + extra_data
+def calculate_chunk(
+    file: io.BufferedIOBase, start_offset: int
+) -> Union[ValidChunk, UnknownChunk]:
 
+    rar_end_offset = calculate_rar_size(file, start_offset)
 
-def calculate_chunk(file: io.BufferedIOBase, start_offset: int) -> Chunk:
-    end_offset = _calculate_end(file, start_offset)
-    return Chunk(start_offset, end_offset)
+    return ValidChunk(
+        start_offset=start_offset,
+        end_offset=rar_end_offset,
+    )
 
 
 def make_extract_command(inpath: str, outdir: str) -> List[str]:
-    return ["unrar", "e", inpath, outdir]
+    return ["unar", inpath, "-o", outdir]
```

Comment:
This will hang waiting for stdin if it's a password protected rar file. Similar to ZIP handling, I'd call it this way:

```
unar -p '' -o outdir inpath
pwprotect.rar: RAR 5
Archive parsing failed! (Missing or wrong password.)
No files extracted.
```